pow/src/main/java/tech/pegasys/artemis/pow/BlockListener.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.pow;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.primitives.UnsignedLong;
+import io.reactivex.disposables.Disposable;
+import java.math.BigInteger;
+import org.apache.tuweni.bytes.Bytes32;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.DefaultBlockParameter;
+import org.web3j.protocol.core.methods.response.EthBlock;
+import tech.pegasys.artemis.pow.event.CacheEth1BlockEvent;
+import tech.pegasys.artemis.util.config.Constants;
+
+public class BlockListener {
+
+  private final Disposable newBlockSubscription;
+
+  public BlockListener(
+      Web3j web3j, EventBus eventBus, DepositContractListener depositContractListener) {
+    this.newBlockSubscription =
+        web3j
+            .blockFlowable(false)
+            .subscribe(
```

Comment:
I agree that we probably should have a more aggressive approach on startup, to make sure we fill the eth1 blocks cache since we might be proposing a block soon (although voting without having seen too many eth1Blocks is not a significant issue for the validator): https://pegasys1.atlassian.net/browse/BC-21.  But for periodic updates, I think subscribing to each block is more straightforward.

Reply:
We can execute periodic logic to pull blocks in `onSlot` for example, which is pretty straightforward.  It's just that we're subscribing to new blocks and putting load on the eth1 node without really needing to.  We really just need a block number to get us started, and then we can pull blocks sequentially.

Reply:
Also - is BC-21 the link you meant to post?